### PR TITLE
[POC] Automate Application update using Updatecli

### PR DESCRIPTION
**Pull request by olblak**
## Description

Automate version bump for applications listed on the Civo Marketplace, using Updatecli.
Updatecli is an open-source declarative dependency management tool, that I started to help me maintaining the Jenkins infrastructure project, and then extended to other projects.

## Context 

My team is maintaining the Epinio project and we would like to automate the version bump on this repository,  each time a new Epinio version is published in the same way that we bump our helm chart, documentation,etc..

For that we use Updatecli which is quite flexible and could also benefits other application listed on this repository.

I reached out to @saiyam1814  and he suggested me to show how to automate four other applications, on top of Epinio:

* Rancher
* Keda
* Argocd
* Cert-manager

You can see the pullrequest open on my fork https://github.com/olblak/kubernetes-marketplace/pulls

I documented in the "README.md", how to use this project and in the "CONTRIBUTING.md" how to add more manifest.
By updating the values.yaml file, we can target a different repository than my fork.

## Disclaimer

Updatecli is a side project that I created to help me maintaining various open source projects that I am working on. While I am doing my best to keep it stable and follow semantic versioning, in the current state, I cannot provide long term guarantee. 

That being said, Updatecli is a command line tool executed from a CI environment. It creates a temporary branch named "updatecli_<pipelineID>", apply the changes on that branch and open a pullrequest, so I think the risk is limited, and it can be disabled at any time.

---
If your pull request is to submit a new application to the marketplace, please answer the following questions:

* [ ] Have you followed the guidelines in our Contributing document in setting up the application?
* [ ] Have you checked that the application your submission is proposing to add does not yet exist on the Marketplace, or as a pull request to be processed?
* [ ] Have you tested your application on the Civo managed Kubernetes service? Please include a screenshot.
* [ ] Can you confirm that the software you are submitting licensed for use in a publicly-available Kubernetes marketplace?
* [ ] Have you included the specific version of the application to be installed?

If your pull request concerns an existing Marketplace application, please make sure you have:

* [ ] Notified the Maintainer of the application in this pull request so that they are aware of your proposal.
* [ ] Outlined the changes you are proposing, and the reasons these are required (e.g. stability, compatibility with new versions of Kubernetes implementations, etc).
* [ ] Tested that the application works with the proposed updates applied (including a screenshot).
* [ ] Updated the version number of the app if that has changed.

### CODE REVIEW

File: updatecli/values.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+github:
+  owner: olblak
```
Comment (olblak) [inline updatecli/values.yaml:2]:
```suggestion
  owner: civo
```

Should be updated to civo for Updatecli to submit changes to this repository.
If Updatecli is executed from a GitHub Action, then we could retrieve the owner and repository from the GitHub Action environment variable.

**Comment by olblak** [comment]
For context I updated the argocd manifest. I recently noticed that the argocd publish different release line on github release, 2.6, 2.5, 2.4, which is not a reliable way to know which one is the Helm chart app version.
So instead I updated the manifest to monitor the chart `appversion` and `version` in the repository argoproj/argo-helm and then run a condition to check if that version is available on the helm repository, to avoid racing issues 
where helm chart files are updated but not published yet.

Commit: https://github.com/civo/kubernetes-marketplace/pull/389/commits/56dd64a6364827d1eae15ba824afb91e6ff1f92c

**Comment by saiyam1814** [comment]
We have some lint requirements for this, we need to skip the lint checking for updatecli directy, else the actions will fail everytime. Right now the link check is to check for manifest file, logo sizing etc. LEt me see how we can fix that first, apart from that this looks good to go right ?

**Comment by olblak** [comment]
Should be good, I tested on https://github.com/olblak/kubernetes-marketplace/actions/runs/4322982621/jobs/7546047192 . Obviously that action fails on the fork as the github_token do not have permission to push changes to civo/kubernetes-marketplace

**Review by saiyam1814 — APPROVED** [review]
LGTM

**Comment by saiyam1814** [comment]
I have merged this, Now its an hourly action run? 
Right now nothing is triggered. @olblak

**Comment by olblak** [comment]
> I have merged this, Now its an hourly action run? 
> Right now nothing is triggered. @olblak 

:tada: yes atm, it's a hourly action run, and it can be manually triggered via GH UI. You could add additional trigger such on branch change. I am super excited to see it working.

**Comment by saiyam1814** [comment]
Yes, its looking good, the PR are now open !

**Pull request merged**